enterprise/views.py

```diff
@@ -582,35 +582,37 @@ def _enroll_learner_in_course(
         if course_id and self.is_course_run_id(course_id):
             if license_uuid:
                 enrollment_api_client = EnrollmentApiClient()
-                course_mode = get_best_mode_from_course_key(course_id)
-                LOGGER.info(
-                    'Retrieved Course Mode: {course_modes} for Course {course_id}'.format(
-                        course_id=course_id,
-                        course_modes=course_mode
-                    )
-                )
-                try:
-                    enrollment_api_client.enroll_user_in_course(
-                        request.user.username,
-                        course_id,
-                        course_mode
-                    )
+                existing_enrollment = enrollment_api_client.get_course_enrollment(request.user.username, course_id)
+                if not existing_enrollment or existing_enrollment.get('mode') == 'audit':
```

Comment:
Nit: I think there's a constant defined somewhere for `audit` mode.

Reply:
^ https://github.com/edx/edx-enterprise/blob/master/enterprise/constants.py#L50

Reply:
Ahhh I had used the hard coded string cuz I saw it elsewhere so I thought that meant edx-enterprise didn't have them as constants. I'll switch over both here and where I saw the other hard coded strings to also use the constants.